src/WebSockets.jl

```diff
@@ -210,12 +214,13 @@ function mask!(to, from, l, mask=rand(UInt8, 4))
     return mask
 end
 
-function Base.close(ws::WebSocket)
+function Base.close(ws::WebSocket; statuscode::Union{Int, Nothing}=nothing)
+    isopen(ws) || return
```

Comment:
I haven't had a chance to think this through properly, but I'm not sure if doing nothing on `!isopen(ws)` is correct. `isopen` is defined in terms of `rxclosed` whereas `closewrite` is concerned with `txclosed`.

Saying `isopen(ws) || return` seems like it might stop the final `WS_CLOSE` being sent when the peer initialises the close.

https://github.com/JuliaWeb/HTTP.jl/blob/e4a133e20e9ee6b4445cafb86193dd490768d228/src/WebSockets.jl#L228

Reply:
Yeah, the reason I put this in was that writing to a closed connection throws `stream is closed or unusable`, but that's like.... 500% accurate.

Reply:
I think I originally thought it would prevent errors when double-closing a connection, but there's already nothing wrong with that scenario.